pytorch/containers/custom_container/task.py

```diff
@@ -0,0 +1,151 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an \"AS IS\" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import torch
+import torch.optim as optim
+import torch.nn as nn
+
+import data_utils
+import model
+
+
+def train(net, train_loader, optimizer, epoch):
+    """Create the training loop"""
+    net.train()
+    criterion = nn.BCELoss()
+    running_loss = 0.0
+
+    for batch_index, data in enumerate(train_loader):
+        features = data['features']
+        target = data['target']
+
+        # zero the parameter gradients
+        optimizer.zero_grad()
+        # forward + backward + optimize
+        outputs = net(features)
+        loss = criterion(outputs, target)
+        loss.backward()
+        optimizer.step()
+
+        # print statistics
+        running_loss += loss.item()
+        if batch_index % 6 == 5:  # print every 6 mini-batches
+            print('[%d, %5d] loss: %.3f' %
+                  (epoch, batch_index + 1, running_loss / 6))
+            running_loss = 0.0
+
+
+def test(net, test_loader):
+    """Test the DNN"""
+    net.eval()
+    criterion = nn.BCELoss()  # https://pytorch.org/docs/stable/nn.html#bceloss
+    test_loss = 0
+    correct = 0
+
+    with torch.no_grad():
+        for i, data in enumerate(test_loader, 0):
+            features = data['features']
+            target = data['target']
+            output = net(features)
+            # Binarize the output
+            pred = output.apply_(lambda x: 0.0 if x < 0.5 else 1.0)
+            test_loss += criterion(output, target)  # sum up batch loss
+            correct += pred.eq(target.view_as(pred)).sum().item()
+
+    test_loss /= len(test_loader.dataset)
+    print('\nTest set:\n\tAverage loss: {:.4f}'.format(test_loss))
+    print('\tAccuracy: {}/{} ({:.0f}%)\n'.format(
+            correct,
+            (len(test_loader) * test_loader.batch_size),
+            100. * correct / (len(test_loader) * test_loader.batch_size)))
+
+
+def train_model(args):
+    """Load the data, train the model, test the model, export / save the model
+    """
+    torch.manual_seed(args.seed)
+
+    # Open our dataset
+    train_loader, test_loader = data_utils.load_data(args.test_split,
+                                                     args.batch_size)
+
+    # Create the model
+    net = model.SonarDNN().double()
+    optimizer = optim.SGD(net.parameters(), lr=args.lr,
+                          momentum=args.momentum, nesterov=False)
+
+    # Train / Test the model
+    for epoch in range(1, args.epochs + 1):
+        train(net, train_loader, optimizer, epoch)
+        test(net, test_loader)
+
+    if args.model_name:
```

Comment:
since `args.model_name` has a default value now, perhaps this first `if` is no longer needed?